### PR TITLE
New Hex package manager Spice

### PR DESCRIPTION
**Pull request by milmazz**
**What does your Instant Answer do?**
Looks up packages on Hex, an Erlang/Elixir package manager

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Allows quicker lookup of Erlang / Elixir packages

**What is the data source for your Instant Answer? (Provide a link if possible)**
https://hex.pm

**Why did you choose this data source?**
It is completely open-source and programmer-oriented, as well as being incredibly user-friendly.

**Is there any other alternative (better) data sources?**
No

**What are some example queries that trigger this Instant Answer?**
```
hex install phoenix
elixir install httpotion
elixir ex_doc
```

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Programmers, hobbyists, sysadmins

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**
No

**Which existing Instant Answers will this one supersede/overlap with?**
None, though I did base it off NPM due to the similar functionality and API

**Are you having any problems? Do you need our help with anything?**
No problems so far.

**What are the terms of use for the API? Will DuckDuckGo need specific authorization (e.g. an API key)? Are there any costs associated with API usage?**
There aren't any I'm aware of. The code for the website is open-source and the contributors, tend to be enthusiasts of open-source projects - I'm sure they won't mind, especially if it benefits them and users of the project.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

screen shot 2016-04-03 at 9 47 23 pm

**Paste your Instant Answer page URL**
https://duck.co/ia/view/hex

### CODE REVIEW

**Comment by daxtheduck** [comment]
@milmazz Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by milmazz** [comment]
Any feedback about this?

**Comment by tagawa** [comment]
@milmazz, thanks again for creating the Hex package manager IA! You’ve been doing a really good job here and we'd like you to help our community tackle the next big challenge - our mission to become the best search engine for programmers: https://duck.co/blog/post/297/help-for-programmers.

When you're ready to move on, please: 

- Create a new Instant Answer to increase our Programming IA coverage

OR

- Test live programming Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage).

Thanks again!

File: lib/DDG/Spice/Hex.pm [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+use DDG::Spice;
+
+spice is_cached => 1;
+spice proxy_cache_valid => "200 1d";
+
+spice wrap_jsonp_callback => 1;
+
+spice to => 'https://hex.pm/api/packages/$1';
+
+triggers startend => 'hex', 'elixir';
```
Comment (moollaza) [inline lib/DDG/Spice/Hex.pm]: the term `hex` may be too vague, I think something more specific like `hex package` would be more appropriate.

For example "hex to decimal conversion", "hex bit driver"

Comment (moollaza) [inline lib/DDG/Spice/Hex.pm]: Elixir seems to have the same problem. As a guitarist Elixir makes me think of the guitar string brand:

hex_install_httpotion_at_duckduckgo

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=hex install httpotion)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/hex_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=hex install httpotion)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/hex_mobile.png)

**Comment by moollaza** [comment]
@milmazz apologies for the delay on getting this reviewed! You've submitted a nearly perfect Pull Request, thank you very much for doing such an excellent job! :smile: 

I've left some comments on the trigger words. Although it's not required, I was wondering what you think about adding a little more content to the body by using the `record` template other information like the maintainer, licenses, downloads etc, to fill out the content a little more?

Some other ideas would be putting a link to the GH repo in the InfoBox, or potentially as a subtitle. As well, mentioning the updated_at somewhere (again subtitle or Infobox)

**Comment by moollaza** [comment]
@milmazz Thanks! This LGTM :+1:

It seems like the API does a good job of weeding out any bad results, we may need to revisit the triggers if it becomes a problem.

Thanks again for doing such an awesome job, it's rare that an Instant Answer is this well made by a first time contributor! I don't think I would have done anything differently if I made it myself :+1:

**Comment by daxtheduck** [comment]
@milmazz this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by milmazz** [comment]
@tagawa Thanks for the information, looking forward to contributing more to with this wonderful community.

@moollaza Thank you for your kind words and your advice. I'll ask for more suggestions from the Elixir community and then I'll update this IA.

**Comment by tagawa** [comment]
@milmazz Sounds good, and well done. Looks like you've really impressed @moollaza with your attention to detail :smile:

**Pull request merged**